corehq/apps/case_search/xpath_functions/value_functions.py

```diff
@@ -61,3 +63,50 @@ def today(domain, node):
     domain_obj = Domain.get_by_name(domain)
     timezone = domain_obj.get_default_timezone() if domain_obj else pytz.UTC
     return datetime.datetime.now(timezone).strftime(ISO_DATE_FORMAT)
+
+
+def date_add(domain, node):
+    from corehq.apps.case_search.dsl_utils import unwrap_value
+
+    assert node.name == 'date_add'
+    if len(node.args) != 3:
+        raise XPathFunctionException(
+            _("The \"date_add\" function expects three arguments"),
```

Comment:
nit: maybe add which arguments it needs?

Reply:
```suggestion
            _("The \"date_add\" function expects three arguments, got {count}".format(
                count=len(node.args),
            )),
```